### ISSUE TITLE
Protect Backup by a strong setup code?

### ISSUE THREAD

**Issue by r10s** [deltachat/deltachat-core#149]
Currently, the exported backups are not protected; instead, before an backup can be exported, the IMAP password must be entered. This way, we make sure, no unauthorized can create a backup.

However, currently, it is unclear, if we leave the existing backup function at all - maybe a "restore from IMAP" is the better way.

But _if_ we keep the backup function in the long term, we should probably switch from the IMAP-password to a strong setup code as already used for the setup messages.

 The process for a backup would be:
- Delta Chat shows the setup code and suggests to write it down
- The user should enter the written setup code - this is to make sure, it was really written down
- On import, eg. after a re-installation, the setup code must be entered

**Comment by STPKITT** [comment]
I'd like to keep the backup function.
I'd prefer though, that the setup code is set by the user instead of Delta Chat dictating a random code.

**Comment by r10s** [comment]
> I'd prefer though, that the setup code is set by the user instead of Delta Chat dictating a random code.

Thank you for your post, I also had this idea.

However, I fear, this would result in a weak protection in most cases which may not be adequate to protect private keys. I do not think, this is a good idea. I think, _if_ we change things here, we should make it right :)

**Comment by Ampli-fier** [comment]
I did "restore from IMAP". One thing came into my mind when using it: one is depending on keys. 
Messages can only be decrypted if keys do match and if they are at hand, whereas the current backup method is independant in this regards.

**Comment by r10s** [comment]
yes, this sounds like a reasonable advantage of keeping the backup function as it is, maybe more secured some day.

**Comment by csb0730** [comment]
I like the current backup function too and used it in the past to correct some errors in database!
But maybe the private keys should be saved and secured separately?

**Comment by testbird** [comment]
The plaintext backup could contain a setup message to store the private keys.

**Comment by r10s** [comment]
of course, setup messages will be included in the backup and one could force the creation of one before exporting. however, this would still require the user to write down the setup code, so, i think, in the end, nothing is won.
